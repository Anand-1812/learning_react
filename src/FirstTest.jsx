import { useState } from "react";


export default function FirstTest() {
  const [heading, setHeading] = useState("Nice heading");

  const clickHandler = () => {
    setHeading("Heading Changes");
  };

  return (
    <>
      <button type="button" onClick={clickHandler}>click me</button>
      <h1>{heading}</h1>
    </>
  );

}

