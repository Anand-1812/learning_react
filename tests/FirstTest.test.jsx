import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import FirstTest from "../src/FirstTest.jsx";

describe("App component", () => {
  it("renders heading through button", () => {
    const { container } = render(<FirstTest />);
    expect(container).toMatchSnapshot();
  });

  it("renders correct heading after click", async () => {
    const user = userEvent.setup();

    render(<FirstTest />);
    const button = screen.getByRole("button", { name: /click me/i });

    await user.click(button);

    const heading = await screen.findByRole("heading");
    expect(heading).toHaveTextContent(/Heading Changes/i);
  });
});

