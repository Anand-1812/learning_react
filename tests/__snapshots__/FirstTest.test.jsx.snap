// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > renders heading through button 1`] = `
<div>
  <button
    type="button"
  >
    click me
  </button>
  <h1>
    Nice heading
  </h1>
</div>
`;
